packages/sidecar/src/messageBuffer.ts

```diff
@@ -60,16 +61,25 @@ export class MessageBuffer<T> {
       atReadPos += 1;
     }
 
-    setTimeout(() => this.stream(readerId, atReadPos), 500);
+    const timeoutKey = setTimeout(() => this.stream(readerId, atReadPos), 500);
```

Comment:
We need a comment here explaining why we are using a timeout before calling stream